plugins/modules/digital_ocean_droplet_info.py

```diff
@@ -0,0 +1,245 @@
+          - m-4vcpu-32gb
+          - c-8
+          - c2-8vcpu-16gb
+          - m3-4vcpu-32gb
+          - g-8vcpu-32gb
+          - so-4vcpu-32gb
+          - m6-4vcpu-32gb
+          - gd-8vcpu-32gb
+          - so1_5-4vcpu-32gb
+          - m-8vcpu-64gb
+          - c-16
+          - c2-16vcpu-32gb
+          - m3-8vcpu-64gb
+          - g-16vcpu-64gb
+          - so-8vcpu-64gb
+          - m6-8vcpu-64gb
+          - gd-16vcpu-64gb
+          - so1_5-8vcpu-64gb
+          - m-16vcpu-128gb
+          - c-32
+          - c2-32vcpu-64gb
+          - m3-16vcpu-128gb
+          - m-24vcpu-192gb
+          - g-32vcpu-128gb
+          - so-16vcpu-128gb
+          - m6-16vcpu-128gb
+          - gd-32vcpu-128gb
+          - m3-24vcpu-192gb
+          - g-40vcpu-160gb
+          - so1_5-16vcpu-128gb
+          - m-32vcpu-256gb
+          - gd-40vcpu-160gb
+          - so-24vcpu-192gb
+          - m6-24vcpu-192gb
+          - m3-32vcpu-256gb
+          - so1_5-24vcpu-192gb
+          - so-32vcpu-256gb
+          - m6-32vcpu-256gb
+          - so1_5-32vcpu-256gb
+        slug: nyc1
+      size:
+        available: true
+        description: Basic
+        disk: 25
+        memory: 1024
+        price_hourly: 0.00744
+        price_monthly: 5.0
+        regions:
+          - ams2
+          - ams3
+          - blr1
+          - fra1
+          - lon1
+          - nyc1
+          - nyc2
+          - nyc3
+          - sfo1
+          - sfo3
+          - sgp1
+          - tor1
+        slug: s-1vcpu-1gb
+        transfer: 1.0
+        vcpus: 1
+      size_slug: s-1vcpu-1gb
+      snapshot_ids: []
+      status: active
+      tags:
+        - tag1
+      vcpus: 1
+      volume_ids: []
+      vpc_uuid: 123-abc-567a
+  type: list
+'''
+
+from traceback import format_exc
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.digitalocean.plugins.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def run(module):
+    rest = DigitalOceanHelper(module)
+
+    if module.params["id"]:
+        path = "droplets/" + module.params["id"]
+        return_codes = (200, 404)
+    else:
+        path = "droplets"
+        return_codes = (200)
+
+    response = rest.get(path)
+
+    if response.status_code not in return_codes:
+        module.fail_json(msg="Failed to fetch 'droplets' information due to error: %s" % response.json["message"])
+
+    if module.params["id"]:
+        data = [response.json["droplet"]]
+    elif module.params["name"]:
+        data = [d for d in response.json["droplets"] if d["name"] == module.params["name"]]
+    else:
+        data = response.json["droplets"]
+
+    module.exit_json(changed=False, data=data)
+
+
+def main():
+    argument_spec = DigitalOceanHelper.digital_ocean_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', required=False, default=None),
+        id=dict(type='str', required=False, default=None)
```

Comment:
The module code treats these two arguments as mutually exclusive (it assumes that at most one will be set), but the parameter validation does not enforce that. So we either need to mark those two as mutually exclusive (and of course also document that fact in the documentation block) or change the run method to be able to handle the situation where both are present.

See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#dependencies-between-module-options for more information on how to specify parameter dependencies.

Reply:
Fixed.